### ISSUE TITLE
🐛 Bug Report: smtp not working

### ISSUE THREAD

**Issue by igorek1955** [appwrite/appwrite#4933]
### 👟 Reproduction steps

deployed project on vps (http://90.156.230.207/)
followed the exact steps from this tutorial:
https://www.youtube.com/watch?v=cCWlYAO192I
and when I try to use any smtp functionality - I don't receive any email. But http response is success:
Token(id=63ad6d2bc76166f6ae8a, createdAt=2022-12-29T10:34:19.818+00:00, userId=63ac6bf4782644122c14, secret=, expire=2022-12-29T11:34:19.816+00:00)

in browser Im also unable to recover password from console

Also I tried using different smtp providers and/or smtp ports, but to no avail 

my .env:
_APP_SMTP_HOST=smtp.mailgun.org
_APP_SMTP_PORT=587
_APP_SMTP_SECURE=tls
_APP_SMTP_USERNAME=[login]
_APP_SMTP_PASSWORD=[password]

doctor report:
Checking for production best practices...
🟢 Hostname has a public suffix (api.orgoazm.ru)
🟢 CNAME target has a public suffix (api.orgoazm.ru)
🔴 Not using a unique secret key for encryption
🟢 App environment is set for production
🟢 Abuse protection is enabled
🟢 Console access limits are enabled
🔴 HTTPS force option is disabled
🔴 Logging adapter is disabled

Checking connectivity...
Database............connected 👍
Queue...............connected 👍
Cache...............connected 👍
SMTP.............disconnected 👎
StatsD..............connected 👍
InfluxDB............connected 👍

Checking volumes...
🟢 Uploads Volume is readable
🟢 Uploads Volume is writeable
🟢 Cache Volume is readable
🟢 Cache Volume is writeable
🟢 Config Volume is readable
🟢 Config Volume is writeable
🟢 Certs Volume is readable
🟢 Certs Volume is writeable

Checking disk space usage...
🟢 Uploads Volume has 24.69GB free space (21.9% used)
🟢 Cache Volume has 24.69GB free space (21.9% used)
🟢 Config Volume has 24.69GB free space (21.9% used)
🟢 Certs Volume has 24.69GB free space (21.9% used)

You are running the latest version of Appwrite! 🥳

### 👍 Expected behavior

should receive emails 

### 👎 Actual Behavior

don't receive any emails in inbox or spam

### 🎲 Appwrite version

Version 1.2.x

### 💻 Operating system

Linux

### 🧱 Your Environment

appwrite version 1.2.0
using android sdk
server running Ubuntu 20.04
1 CPU • 2 gb RAM • 30 gb SSD

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
@igorek1955, some questions for you:

1. Is there any error in the `appwrite-worker-mails` container?
2. Are you able to test SMTP on the server with some other tool?
3. Who is your VPS provider?

**Comment by Maxbrt18** [comment]
Same error for me, I have this error : `{message: SMTP Disabled, code: 503, type: general_smtp_disabled, version: 1.2.0}`
and `docker exec appwrite doctor` shows : 

 ```
 __   ____  ____  _  _  ____  __  ____  ____     __  __  
 / _\ (  _ \(  _ \/ )( \(  _ \(  )(_  _)(  __)   (  )/  \ 
/    \ ) __/ ) __/\ /\ / )   / )(   )(   ) _)  _  )((  O )
\_/\_/(__)  (__)  (_/\_)(__\_)(__) (__) (____)(_)(__)\__/ 

👩‍⚕️ Running Appwrite Doctor for version 1.2.0 ...

Checking for production best practices...
🟢 Hostname has a public suffix (backend.kholle.fr)
🟢 CNAME target has a public suffix (backend.kholle.fr)
🟢 Using a unique secret key for encryption
🟢 App environment is set for production
🟢 Abuse protection is enabled
🟢 Console access limits are enabled
🔴 HTTPS force option is disabled
🔴 Logging adapter is disabled

Checking connectivity...
Database............connected 👍
Queue...............connected 👍
Cache...............connected 👍
SMTP.............disconnected 👎
StatsD..............connected 👍
InfluxDB............connected 👍

Checking volumes...
🟢 Uploads Volume is readable
🟢 Uploads Volume is writeable
🟢 Cache Volume is readable
🟢 Cache Volume is writeable
🟢 Config Volume is readable
🟢 Config Volume is writeable
🟢 Certs Volume is readable
🟢 Certs Volume is writeable

Checking disk space usage...
🟢 Uploads Volume has 38.88GB free space (19.46% used)
🟢 Cache Volume has 38.88GB free space (19.46% used)
🟢 Config Volume has 38.88GB free space (19.46% used)
🟢 Certs Volume has 38.88GB free space (19.46% used)

You are running the latest version of Appwrite! 🥳
```

The strange thing is that `HTTPS force option is disabled` is actually enabled in my .env file, same for SMTP which is configured correctly. I think the problem is that the .env file is not read.

**Comment by igorek1955** [comment]
> @igorek1955, some questions for you:
> 
> 1. Is there any error in the `appwrite-worker-mails` container?
> 2. Are you able to test SMTP on the server with some other tool?
> 3. Who is your VPS provider?

1 - The error turns up in log after sometime after I click recover password from console . Here's the contents:
[critical] (Job{v1-mails} | ID: 296b55a5d1596a3ee82a4b65ebdac869 | MailsV1 | [{"project":{"$id":"console","$internalId":"console","name":"Appwrite","$collection":"projects","description":"Appwrite core engine","logo":"","teamId":-1,"webhooks":[],"keys":[],"platforms":[{"$collection":"platforms","name":"Localhost","type":"web","hostname":"localhost"},{"$collection":"platforms","name":"Current Host","type":"web","hostname":"90.156.230.207"}],"legalName":"","legalCountry":"","legalState":"","legalCity":"","legalAddress":"","legalTaxId":"","auths":{"limit":1,"duration":31536000},"authWhitelistEmails":[],"authWhitelistIPs":[]},"user":{"$id":"","$collection":"users"},"payload":[],"recipient":"igorek1955@gmail.com","url":"http:\/\/90.156.230.207\/recover?userId=63ac17bca91d3da6fa1f&secret=87396ef55fb104c8824d3c31b3018c0aa117bd63981bf8ffcf86af0d1e4636b8413d7960abefc810f829af456457b966462704d64f475666a037b9717faf626b235c99161d3b9a6c03ef881874b22768c7125ffbfd34255831b930242824ccd8582182ae3caa3fa49cdafbf5285374f67013deefb1485bc27c4fd4d6b85801a8&expire=2022-12-30+18%3A05%3A01.488","locale":"en","type":"recovery","name":"igor","team":null,"events":["."]}]) has failed Exception: Error sending mail: SMTP Error: Could not connect to SMTP host. in /usr/src/code/app/workers/mails.php:126

2 - I tested smtp on appwrite server by clicking "recover" from the console login page , and I tested it by sending email verifications using appwrite android sdk. But I tested my SMTP provider via https://www.gmass.co/smtp-test

3 - timeweb.ru , its also my smtp provider (I also tried mailgun and sendgrid)

**Comment by igorek1955** [comment]
also, on somewhat related topic, how do I edit email templates?

**Comment by idoRosen25** [comment]
related to this issue. im having a similar issue with using sendGrid but my error is that the authentication fails (although i followed all the instructions on their website and in this link `https://gist.github.com/eldadfux/cbc4c0664d08b36dfe90f83748b242d9` correctly).
any thoughts on how to fix the authentication issue? do you recommend using another provider for SMTP server?

**Comment by stnguyen90** [comment]
> The strange thing is that `HTTPS force option is disabled` is actually enabled in my .env file, same for SMTP which is configured correctly. I think the problem is that the .env file is not read.

@Maxbrt18, does `docker compose exec appwrite-worker-mails vars` show the correct values?

**Comment by stnguyen90** [comment]
> authentication fails

@idoRosen25, does `docker compose exec appwrite-worker-mails vars` show the correct values? It should be `SG.something.something`.

**Comment by Maxbrt18** [comment]
> 



> > The strange thing is that `HTTPS force option is disabled` is actually enabled in my .env file, same for SMTP which is configured correctly. I think the problem is that the .env file is not read.
> 
> @Maxbrt18, does `docker compose exec appwrite-worker-mails vars` show the correct values?

After downgrading to 1.1.2 and then back to 1.2.0 the problem seems to be gone.

**Comment by idoRosen25** [comment]
> docker compose exec appwrite-worker-mails vars

yes. it shows the correct information but still shows authentication error.

**Comment by idoRosen25** [comment]
> > 
> 
> > > The strange thing is that `HTTPS force option is disabled` is actually enabled in my .env file, same for SMTP which is configured correctly. I think the problem is that the .env file is not read.
> > 
> > 
> > @Maxbrt18, does `docker compose exec appwrite-worker-mails vars` show the correct values?
> 
> After downgrading to 1.1.2 and then back to 1.2.0 the problem seems to be gone.

can you state the steps you did for that? might solve my issue as well.

**Comment by gepd** [comment]
This could be related to this issue: https://github.com/appwrite/appwrite/issues/4952

**Comment by Maxbrt18** [comment]
> > > 
> 
> > 
> 
> > > > The strange thing is that `HTTPS force option is disabled` is actually enabled in my .env file, same for SMTP which is configured correctly. I think the problem is that the .env file is not read.
> 
> > > 
> 
> > > 
> 
> > > @Maxbrt18, does `docker compose exec appwrite-worker-mails vars` show the correct values?
> 
> > 
> 
> > After downgrading to 1.1.2 and then back to 1.2.0 the problem seems to be gone.
> 
> 
> 
> can you state the steps you did for that? might solve my issue as well.

I just downgraded to 1.1.2 with this command (because mails worked with this version):

```
docker run -it --rm \
    --volume /var/run/docker.sock:/var/run/docker.sock \
    --volume "$(pwd)"/appwrite:/usr/src/code/appwrite:rw \
    --entrypoint="install" \
    appwrite/appwrite:1.1.2

```

After a while, I wanted to try to see if the error persisted and I just installed the last version with : 

```

docker run -it --rm \
    --volume /var/run/docker.sock:/var/run/docker.sock \
    --volume "$(pwd)"/appwrite:/usr/src/code/appwrite:rw \
    --entrypoint="install" \
    appwrite/appwrite:1.2.0
```

Hope it will solve your problem

**Comment by stnguyen90** [comment]
@idoRosen25, did you also set the `_APP_SYSTEM_EMAIL_ADDRESS` env var and go through the [Sender Authentication](https://docs.sendgrid.com/glossary/sender-authentication) steps in SendGrid?

**Comment by idoRosen25** [comment]
Yes. Though it doesn't state in their integration steps that it is needed

**Comment by igorek1955** [comment]
> @igorek1955
> 
> > SMTP Error: Could not connect to SMTP host
> 
> Who is your VPS provider? Your VPS provider could be blocking the SMTP port. Please try testing on the server rather than on 3rd party website.
> 
> > how do I edit email templates?
> 
> Currently, you'd have to mount in a custom version of [email-base.tpl](https://github.com/appwrite/appwrite/blob/master/app/config/locale/templates/email-base.tpl).

my vps provider is timeweb. I tried launching appwrite on localhost and smtp was working fine, so it seems that the problem lies with vps. How do I troubleshoot it exactly?

**Comment by idoRosen25** [comment]
> > > >
> > 
> > 
> > > 
> > 
> > 
> > > > > The strange thing is that `HTTPS force option is disabled` is actually enabled in my .env file, same for SMTP which is configured correctly. I think the problem is that the .env file is not read.
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > @Maxbrt18, does `docker compose exec appwrite-worker-mails vars` show the correct values?
> > 
> > 
> > > 
> > 
> > 
> > > After downgrading to 1.1.2 and then back to 1.2.0 the problem seems to be gone.
> > 
> > 
> > can you state the steps you did for that? might solve my issue as well.
> 
> I just downgraded to 1.1.2 with this command (because mails worked with this version):
> 
> ```
> docker run -it --rm \
>     --volume /var/run/docker.sock:/var/run/docker.sock \
>     --volume "$(pwd)"/appwrite:/usr/src/code/appwrite:rw \
>     --entrypoint="install" \
>     appwrite/appwrite:1.1.2
> ```
> 
> After a while, I wanted to try to see if the error persisted and I just installed the last version with :
> 
> ```
> 
> docker run -it --rm \
>     --volume /var/run/docker.sock:/var/run/docker.sock \
>     --volume "$(pwd)"/appwrite:/usr/src/code/appwrite:rw \
>     --entrypoint="install" \
>     appwrite/appwrite:1.2.0
> ```
> 
> Hope it will solve your problem

did your existing data get affected by downgrading the appwrite version? (DB, CF, etc.)

**Comment by Maxbrt18** [comment]
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > > > The strange thing is that `HTTPS force option is disabled` is actually enabled in my .env file, same for SMTP which is configured correctly. I think the problem is that the .env file is not read.
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > > @Maxbrt18, does `docker compose exec appwrite-worker-mails vars` show the correct values?
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > After downgrading to 1.1.2 and then back to 1.2.0 the problem seems to be gone.
> 
> > > 
> 
> > > 
> 
> > > can you state the steps you did for that? might solve my issue as well.
> 
> > 
> 
> > I just downgraded to 1.1.2 with this command (because mails worked with this version):
> 
> > 
> 
> > ```
> 
> > docker run -it --rm \
> 
> >     --volume /var/run/docker.sock:/var/run/docker.sock \
> 
> >     --volume "$(pwd)"/appwrite:/usr/src/code/appwrite:rw \
> 
> >     --entrypoint="install" \
> 
> >     appwrite/appwrite:1.1.2
> 
> > ```
> 
> > 
> 
> > After a while, I wanted to try to see if the error persisted and I just installed the last version with :
> 
> > 
> 
> > ```
> 
> > 
> 
> > docker run -it --rm \
> 
> >     --volume /var/run/docker.sock:/var/run/docker.sock \
> 
> >     --volume "$(pwd)"/appwrite:/usr/src/code/appwrite:rw \
> 
> >     --entrypoint="install" \
> 
> >     appwrite/appwrite:1.2.0
> 
> > ```
> 
> > 
> 
> > Hope it will solve your problem
> 
> 
> 
> did your existing data get affected by downgrading the appwrite version? (DB, CF, etc.)


Nothing was deleted, everything seems to be working fine, but I made a backup to be sure nothing would be affected.

**Comment by idoRosen25** [comment]
> docker exec appwrite doctor

@Maxbrt18  i tried to downgrade but it still not working. im still getting en error saying it could not authenticate.
did you get it to work on the downgraded version before upgrading again?

**Comment by Maxbrt18** [comment]
> > docker exec appwrite doctor
> 
> 
> 
> @Maxbrt18  i tried to downgrade but it still not working. im still getting en error saying it could not authenticate.
> 
> did you get it to work on the downgraded version before upgrading again?
> 
> did you also change the APP_SYSTEM_EMAIL_ADDRESS to be the same as in the smtp sender info?
> 
> for some reason all my attempts are failing with could not authenticate error.
> 
> 
> 
> I updated the APP_SMTP vars as described in sendGrid integration steps. also tried using my actual login info as username/password and changing the APP_SYSTEM_EMAIL_ADDRESS but with no success.
> 
> (im testing on version 1.1.2)

My problem wasn't about smtp authentication but about smtp that was disabled. After downgrading, it worked fine but I had a translation error (fixed in 1.2.0) so I needed to upgrade to 1.2.0. And after upgrading everything worked as expected. For information, my APP_SYSTEM_EMAIL_ADDRESS is the default one.

**Comment by stnguyen90** [comment]
> my vps provider is timeweb.ru. I tried launching appwrite on localhost and smtp was working fine, so it seems that the problem lies with vps. How do I troubleshoot it exactly?

@igorek1955, one common problem is VPS providers block outgoing SMTP ports to protect against spam. So, you'll need to find some way to test the SMTP settings **from your server**. For example, you can do `telnet smtp.mailgun.org 587`. If that doesn't show `Connected`, there might be some blocking in place.

I'm going to close this issue because SMTP is working fine and there's too many threads goings on. If you need additional help, please check the [docs](https://appwrite.io/docs/email-delivery) and/or join us on [Discord](https://appwrite.io/discord).
